pkg/controller/machine/controller.go

```diff
@@ -82,6 +109,47 @@ func newReconciler(mgr manager.Manager, actuator Actuator) reconcile.Reconciler
 	return r
 }
 
+func stringPointerDeref(stringPointer *string) string {
+	if stringPointer != nil {
+		return *stringPointer
+	}
+	return ""
+}
+
+func predicateUpdateFunc(e event.UpdateEvent) bool {
+	oldObject, ok := e.ObjectOld.(*machinev1.Machine)
+	if !ok {
+		// pass through
+		return true
+	}
+	newObject, ok := e.ObjectNew.(*machinev1.Machine)
+	if !ok {
+		// pass through
+		return true
+	}
+
+	// Don't queue when phasing so we have control over
+	// reconcile.Result{Requeue: true, RequeueAfter: retryTime, err}
+	oldPhase := stringPointerDeref(oldObject.Status.Phase)
+	newPhase := stringPointerDeref(newObject.Status.Phase)
+	if oldPhase != newPhase {
+		oldObjectCopy := oldObject.DeepCopy()
+		newObjectCopy := newObject.DeepCopy()
+		// reset phasing values to be ignored when comparing
+		oldObjectCopy.Status.Phase = nil
+		oldObjectCopy.Status.LastUpdated = nil
+		newObjectCopy.Status.Phase = nil
+		newObjectCopy.Status.LastUpdated = nil
+		if !equality.Semantic.DeepEqual(oldObjectCopy, newObjectCopy) {
```

Comment:
Would we need to ignore the resource version as well? This is modeled loosely on the pod controller, right? Does it do something similar?